python_modules/dagster/dagster/_core/definitions/decorators/asset_decorator.py

```diff
@@ -306,184 +306,153 @@ def resolve_asset_key_and_name_for_decorator(
-        output_required: bool = True,
-        freshness_policy: Optional[FreshnessPolicy],
-        auto_materialize_policy: Optional[AutoMaterializePolicy],
-        backfill_policy: Optional[BackfillPolicy],
-        retry_policy: Optional[RetryPolicy],
-        code_version: Optional[str],
-        key: Optional[CoercibleToAssetKey],
-        check_specs: Optional[Sequence[AssetCheckSpec]],
-        owners: Optional[Sequence[str]],
-    ):
-        self.name = name
-        self.key_prefix = key_prefix
-        self.ins = ins
-        self.deps = deps
-        self.metadata = metadata
-        self.tags = tags
-        self.description = description
-        self.required_resource_keys = required_resource_keys
-        self.io_manager_key = io_manager_key
-        self.io_manager_def = io_manager_def
-        self.config_schema = config_schema
-        self.compute_kind = compute_kind
-        self.dagster_type = dagster_type
-        self.partitions_def = partitions_def
-        self.op_tags = op_tags
-        self.resource_defs = resource_defs
-        self.group_name = group_name
-        self.output_required = output_required
-        self.freshness_policy = freshness_policy
-        self.retry_policy = retry_policy
-        self.auto_materialize_policy = auto_materialize_policy
-        self.backfill_policy = backfill_policy
-        self.code_version = code_version
-        self.check_specs = check_specs
-        self.key = key
-        self.owners = owners
-
-    def __call__(self, fn: Callable[..., Any]) -> AssetsDefinition:
-        from dagster._config.pythonic_config import (
-            validate_resource_annotated_function,
-        )
-        from dagster._core.execution.build_resources import wrap_resources_for_execution
-
-        validate_resource_annotated_function(fn)
+class AssetArgs(NamedTuple):
+    required_resource_keys: AbstractSet[str]
+    name: Optional[str]
+    key_prefix: Optional[CoercibleToAssetKeyPrefix]
+    ins: Mapping[str, AssetIn]
+    deps: Iterable[AssetDep]
+    metadata: Optional[ArbitraryMetadataMapping]
+    tags: Optional[Mapping[str, str]]
+    description: Optional[str]
+    config_schema: Optional[UserConfigSchema]
+    resource_defs: Dict[str, object]
+    io_manager_key: Optional[str]
+    io_manager_def: Optional[object]
+    compute_kind: Optional[str]
+    dagster_type: Optional[DagsterType]
+    partitions_def: Optional[PartitionsDefinition]
+    op_tags: Optional[Mapping[str, Any]]
+    group_name: Optional[str]
+    output_required: bool
+    freshness_policy: Optional[FreshnessPolicy]
+    auto_materialize_policy: Optional[AutoMaterializePolicy]
+    backfill_policy: Optional[BackfillPolicy]
+    retry_policy: Optional[RetryPolicy]
+    code_version: Optional[str]
+    key: Optional[CoercibleToAssetKey]
+    check_specs: Optional[Sequence[AssetCheckSpec]]
+    owners: Optional[Sequence[str]]
+
+
+def invoke(args: AssetArgs, fn: Callable[..., Any]) -> AssetsDefinition:
```

Comment:
I'm having a little trouble connecting the name of this function to what it does. Without more context about what's going on, I would expect a function named `invoke` that accepts a `Callable` to invoke that callable.

Thoughts on something verbose like `create_assets_def_from_fn_and_decorator_args`?

Reply:
For good reason because `invoke` is a terrible name